src/chart/generateCategoricalChart.tsx

```diff
@@ -821,6 +822,7 @@ export interface CategoricalChartState {
 
   legendBBox?: DOMRect | null;
 
+  prevDataKey?: DataKey<any>;
```

Comment:
Do we need to use any? 🤔

Reply:
I had the same though, I really dislike using `any`, and when I looked for `DataKey` usages, they were all using it like this. 😅

Reply:
yeah they're all like this right now unfortunately

That being said this is a great fix, we can address the DataKey issue with some of our refactorings that have been happening across the lib